drivers/hwinfo/hwinfo_sam.c

```diff
@@ -74,7 +75,7 @@ static int hwinfo_sam_init(const struct device *arg)
 	/* Disable code loop optimization and sequential code optimization. */
 	fmr = efc->EEFC_FMR;
 
-#ifndef CONFIG_SOC_SERIES_SAM3X
+#if !IS_ENABLED(CONFIG_SOC_SERIES_SAM3U) && !IS_ENABLED(CONFIG_SOC_SERIES_SAM3X)
```

Comment:
Analogues to the discussion in #53200
```suggestion
#if !defined(CONFIG_SOC_SERIES_SAM3U) && !defined(CONFIG_SOC_SERIES_SAM3X)
```

Reply:
Thanks. Updated.